src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -104,7 +98,16 @@ public ConfigurationItemFactory(params Assembly[] assemblies)
         /// <summary>
         /// Gets or sets default singleton instance of <see cref="ConfigurationItemFactory"/>.
         /// </summary>
-        public static ConfigurationItemFactory Default { get; set; }
+        public static ConfigurationItemFactory Default
+        {
+            get
+            {
+                if (defaultInstance == null)
```

Comment:
The first commit of the pull request ("Fixed Auto-loading logs did not render because extension DLL's were loaded before InternalLogger config was read") contains all the changes that were needed to fix the auto-loading logs that did not appear.

This part, in particular, implements lazy instantiation on the ConfigurationItemFactory, because it was created too early. Now the ConfigurationItemFactory is created only after the internal logger had been configured, so that the auto-loading logs actually appear in the internal log file.